plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -277,13 +277,23 @@ public ScheduledJobParser getJobParser() {
 
   @Override
   public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
-    return singletonList(
+    // The worker pool is the primary pool where most of the work is done. The background thread
+    // pool is a separate queue for asynchronous requests to other nodes. We keep them separate to
+    // prevent deadlocks during async fetches on small node counts. Tasks in the background pool
+    // should do no work except I/O to other services.
+    return List.of(
         new FixedExecutorBuilder(
             settings,
-            AsyncRestExecutor.SQL_WORKER_THREAD_POOL_NAME,
+            SQL_WORKER_THREAD_POOL_NAME,
             OpenSearchExecutors.allocatedProcessors(settings),
             1000,
-            null));
+            "thread_pool." + SQL_WORKER_THREAD_POOL_NAME),
+        new FixedExecutorBuilder(
```

Comment:
How would it be different if we use the same thread pool or increase the size of the same pool and submit the task to same.


Threads from all the pools on the node will take turns for the same cores available.

`Now double the number of threads from the sql plugin will be contending for the cpu.`  Just food for thought: is it good or bad?

Now that PPL is becoming front and center in discover I don't mind doing this.

Reply:
The background threads have very light work, it's just serialization costs. They'll take virtually no resources compared to the search and worker threads. Modern threads are cheap so they'll take virtually no resources while waiting for the cluster to reply.